async-nats/src/jetstream/stream.rs

```diff
@@ -91,6 +94,50 @@ impl Stream {
 
         Ok(Consumer::new(T::try_from_consumer_config(info.config)?))
     }
+
+    pub async fn get_or_create_consumer<T: FromConsumer + IntoConsumerConfig>(
+        &self,
+        name: &str,
+        config: T,
+    ) -> Result<Consumer<T>, Error> {
+        let subject = format!(
+            "{}.CONSUMER.INFO.{}.{}",
+            self.prefix, self.info.config.name, name
+        );
+
+        match self.context.request(subject, &json!({})).await? {
+            Response::Err { error } if error.code == 404 => self
```

Comment:
Sure this isn't 503? test for unhappy paths.

```suggestion
            Response::Err { error } if error.code == 404 => self
```

Reply:
yeah, tests cover this case.